configuration/forms.md

```diff
@@ -46,6 +46,26 @@ Used for SMS interfaces such as formatted SMS, SIM applications, and Medic Colle
   }
 }
 ```
+## JSON Form Schema
+| property | type | description | required |
+|---|---|---|---|
+| `meta` | object | Information about the report | yes |
+| `meta.code` | string | The unique form identifier, which will be sent with all reports of this form | yes |
+| `meta.icon` | string | Name of the icon resource shown in the app. | no |
+| `meta.translation_key` | string | Name of the form shown in the app. | no |
+| `fields`| object | Collection of field objects included in the form. | yes |
+| `fields.${field}` | object | Field that is part of the form. | yes |
+| `fields.${field}.type` | string | Data type of field. See [supported data types](#supported-data-types). | yes |
+| `fields.${field}.labels` | object | | no |
+| `fields.${field}.labels.short` | string, object with `translation_key` field, or translation object | Label shown for field in the app, seen when report is viewed in Reports tab. If missing, label will default to a translation key of `report.${form_name}.${field_name}` (eg `report.f.patient_id`) which can be translated in the app languages. | no |
+| `fields.${field}.labels.tiny` | string | Unique identifier within the form for this field. Used in form submission to bind values to fields. Not required for all submission formats. | no |
+| `fields.${field}.position` | integer | Zero based order of this field for incoming reports. | no |
+| `fields.${field}.flags` | object | Additional instructions that could be used by form renderers. For instance `{ "input_digits_only": true }` indicated to SIM applications to show the number keyboard. Obsolete. | no |
+| `fields.${field}.length` | array with two integers | Inclusive range accepted for length of the field. | no |
+| `fields.${field}.required` | boolean | Determines if a report without this field is considered valid. | no |
+| `public_form` | boolean | Determines if reports will be accepted from phone numbers not associated to a contact. Set to false if you want to reject reports from unknown senders. Default: true | no |
+| `facility_reference` | string | The form field whose value is to be used to match the incoming report to a contact's `rc_code`. Useful when reports are sent on behalf of a facility by unknown or various phone numbers. | no |
```

Comment:
Added comment below on what `update_clinics` does when contact is matched with `rc_code` and sender phone doesn't match contact.

Reply:
It looks like the intention is to have CHWs inform the app that they have changed their phone number instead of sending reports on their behalf.

Reply:
I had never realized it updated the contact's phone number. It seems as it was done as such 
6 years ago to [keep track of the most recent phone number for the clinic](https://github.com/medic/medic/blame/7c0404d88d839503b855ad30c48fa701fee69416/sentinel/src/transitions/update_clinics.js#L102-L104). I think predates our current contact system. Back then the contacts were basically a table with only one person and phone number per place, so this use would have made more sense. 

Is this currently putting a `phone` field on the _place_ now... in other words, is the `contact` a place or a person? If it a place, that may need to be revisited since we now associate the phone number to the people, of which one can be the primary contact. Let's discuss to see if we should remove this "feature" and it's related code.

Reply:
The `contact` is a person (if the `refid` corresponds to a clinic, the contact will be the clinics's associated contact).
And I just found a bug: when the contact is updated with the new phone, it's not minified before saving :( . 

If this feature is outdated, happy to remove the code, otherwise we need to add minifying before saving.